pykafka/protocol.py

```diff
@@ -794,17 +814,45 @@ def _unpack_message_set(self, buff, partition_id=-1):
         for message in message_set.messages:
             if message.compression_type == CompressionType.NONE:
                 output.append(message)
+                continue
             elif message.compression_type == CompressionType.GZIP:
                 decompressed = compression.decode_gzip(message.value)
-                output += self._unpack_message_set(decompressed,
-                                                   partition_id=partition_id)
+                messages = self._unpack_message_set(decompressed,
+                                                    partition_id=partition_id)
             elif message.compression_type == CompressionType.SNAPPY:
                 decompressed = compression.decode_snappy(message.value)
-                output += self._unpack_message_set(decompressed,
-                                                   partition_id=partition_id)
+                messages = self._unpack_message_set(decompressed,
+                                                    partition_id=partition_id)
+            if messages[-1].offset < message.offset:
+                # With protocol 1, offsets from compressed messages start at 0
+                assert messages[0].offset == 0
+                delta = message.offset - len(messages) + 1
```

Comment:
I don't follow this math. From my understanding, messages nested in compressed messagesets come back with their offsets starting from 0. Thus if `message.offset` here were 50, I'd expect the first item in `messages` to have its offset set to 51 - the next offset. However, assuming that `len(messages)` here is 5, for example, this math results in `delta == 50 - 5 + 1 == 46`, meaning that below, the message offsets will start from 46 instead of 0. What am I missing here?